qemu/qmp.go

```diff
@@ -928,6 +929,16 @@ func (q *QMP) ExecutePCIVFIODeviceAdd(ctx context.Context, devID, bdf, addr, bus
 		"host":   bdf,
 		"addr":   addr,
 	}
+
+	if runtime.GOARCH == "s390x" {
```

Comment:
Again, for consistency with the `qemu*.go` files, I'd rework this as follows:

- Create `qmp_amd64.go` and `qmp_s390x.go` containing their own implementations of:
  ```
  var archPCIVFIODeviceAddMap = map[string]interface{}{ ... }
  ```
- Make `qmp.go` reference the right implementation. Something like:

  ```
  var pciVFIODeviceAddMap = archPCIVFIODeviceAddMap

  func (q *QMP) ExecutePCIVFIODeviceAdd(...) {
        arg := archPCIVFIODeviceAddMap
        return q.executeCommand(ctx, "device_add", args, nil)
  }
  ```